src/ibutton.toit

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2023 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the LICENSE file.
+
+import gpio
+import one_wire
+
+/**
+An iButton reader.
+
+This class is a small wrapper around the 1-wire protocol.
+
+It assumes that the reader is the only physical device that is attached
+  to the bus. If the reader is not the only device, use the one_wire.Bus
+  class directly.
+*/
+class Reader:
+  bus_/one_wire.Bus
+
+  /**
+  Constructs a reader instance on the 1-wire bus on the given $pin.
+  */
+  constructor pin/gpio.Pin:
+    bus_ = one_wire.Bus pin
+
+  /**
+  Scans the reader for an iButton.
```

Comment:
Something about how long the scan takes and when/why it returns null?

Reply:
done.